apps/vthread_props/src/main/java/vthread_props/Main.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2025, Red Hat Inc. All rights reserved.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package vthread_props;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class Main {
+    public static void main(String[] args) {
+        final String r = String.format("%s %s %s %s %s %s",
+                runTest("System Properties Access", Main::testSystemPropertiesAccess),
+                runTest("LocalDateTime.now()", Main::testLocalDateTimeNow),
+                runTest("ZoneId Access", Main::testZoneIdAccess),
+                runTest("Multiple Time Operations", Main::testMultipleTimeOperations),
+                runTest("Virtual Thread Properties", Main::testVirtualThreadProperties),
+                runTest("Process and System Operations", Main::testProcessAndSystemOperations));
+        System.out.println("\n=== RESULT: " + r + " ===");
+    }
+
+    private static boolean testSystemPropertiesAccess() throws Exception {
+        final CountDownLatch l = new CountDownLatch(1);
+        final AtomicReference<Properties> p = new AtomicReference<>();
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        final Thread vt = Thread.ofVirtual().start(() -> {
+            try {
+                System.out.println("Virtual: " + (Thread.currentThread().isVirtual() ? "True" : "ERROR"));
```

Comment:
The `True vs ERROR` printing here is a bit strange. I assume you do it that way so that log parsing will catch the "error" is that right?

If so I would prefer an `if-statement` controlling for that case and printing the Error if `isVirtual` is `false`. That would remove any potential ambiguity about what's causing this "ERROR".

Reply:
Ack.